web/src/features/auth/slices/authenticationSlice.ts

```diff
@@ -67,6 +71,20 @@ const {
 
 export default authSlice.reducer
 
+export const decodeRoles = (token: string | undefined) => {
+  if (isUndefined(token)) {
+    return []
+  }
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const decodedToken: any = jwt_decode(token)
+  try {
+    return decodedToken.resource_access['wps-web'].roles
```

Comment:
Stupid question - but what happens if the token is expired? Could you have a case where an expired token is processed, and the role is extracted?

Reply:
I'm not sure it's possible since we require authentication before showing anything. But either way, when the frontend features are implemented they should check for `isAuthenticated` as well as roles.